flytekit/engines/flyte/engine.py

```diff
@@ -107,6 +108,21 @@ def fetch_task(self, task_id):
             insecure=_platform_config.INSECURE.get()
         ).client.get_task(task_id)
 
+    def fetch_latest_task(self, named_task):
```

Comment:
Would it be better we put this logic into `fetch_task`? E.g. if there is no `version` specified, we fetch the latest one.

Reply:
Or should we at least stick to `Identifier` instead of `NamedEntityIdentifier` at this layer? Sort of like what `fetch_launch_plan` does.

Reply:
I think your way is correct just to prevent accidental problematic patterns from emerging. Launch plan has the special case because there is a concept of 'active' which is basically a way of indicating what is the currently 'deployed entrypoint' version to a workflow. This makes it easy to indicate which schedule should be active, and write command line tools to kick off workflows.

Tasks don't currently have the active concept and fetch is instead used to create workflows. Creating a workflow with a loose reference _can_ be problematic if you're not prepared for what that entails. Because then registering a workflow isn't always idempotent--if the task version received on different attempts at registration of the workflow, the registration will fail.

So this is pretty explicit and I think it is good for now. If we want to elevate tasks to have a concept like 'active', this change doesn't preclude that and we can do it later--but that is a larger change.

Reply:
Thanks for the detailed explanation. I totally agree.